src/mqt/benchviewer/backend.py

```diff
@@ -23,256 +23,544 @@
+            {
+                "name": "Efficient SU2 ansatz with Random Parameters",
+                "id": "18",
+                "filename": "su2random",
+            },
+            {
+                "name": "Real Amplitudes ansatz with Random Parameters",
+                "id": "19",
+                "filename": "realamprandom",
+            },
+            {
+                "name": "Two Local ansatz with Random Parameters",
+                "id": "20",
+                "filename": "twolocalrandom",
+            },
+            {"name": "W-State", "id": "21", "filename": "wstate"},
+        ]
+
+        self.nonscalable_benchmarks = [
+            {"name": "Ground State", "id": "22", "filename": "groundstate"},
+            {"name": "HHL", "id": "23", "filename": "hhl"},
+            {"name": "Pricing Call Option", "id": "24", "filename": "pricingcall"},
+            {"name": "Pricing Put Option", "id": "25", "filename": "pricingput"},
+            {"name": "Routing", "id": "26", "filename": "routing"},
+            {"name": "Shor's", "id": "27", "filename": "shor"},
+            {"name": "Travelling Salesman", "id": "28", "filename": "tsp"},
+        ]
+
+        self.database: pd.DataFrame | None = None
+        self.mqtbench_all_zip: ZipFile | None = None
+
+    def filter_database(self, filter_criteria: tuple):  # noqa: PLR0912
+        """Filters the database according to the filter criteria.
+
+        Keyword arguments:
+        filterCriteria -- list of all filter criteria
+        database -- database containing all available benchmarks
+
+        Return values:
+        db_filtered["path"].to_list() -- list of all file paths of the selected benchmark files
+        """
+        colnames = [
+            "benchmark",
+            "num_qubits",
+            "indep_flag",
+            "nativegates_flag",
+            "mapped_flag",
+            "compiler",
+            "compiler_settings",
+            "gate_set",
+            "target_device",
+            "path",
+        ]
+        db_filtered = pd.DataFrame(columns=colnames)
+        db_filtered["indep_flag"] = db_filtered["indep_flag"].astype(bool)
+        db_filtered["nativegates_flag"] = db_filtered["nativegates_flag"].astype(bool)
+        db_filtered["mapped_flag"] = db_filtered["mapped_flag"].astype(bool)
+        if len(self.database) == 0:
+            return []
 
+        (
+            (min_qubits, max_qubits),
+            indices_benchmarks,
+            (indep_qiskit_compiler, indep_tket_compiler),
+            (
+                (nativegates_qiskit_compiler, nativegates_tket_compiler),
+                native_qiskit_opt_lvls,
+                native_gatesets,
+            ),
+            (
+                (mapped_qiskit_compiler, mapped_tket_compiler),
+                (mapped_qiskit_opt_lvls, mapped_tket_placements),
+                mapped_devices,
+            ),
+        ) = filter_criteria
```

Comment:
I suppose you are doing something about this in the mypy PR. Filter criteria should definitely be a data class.

Reply:
This is fixed in #190.